Content.Client/Disposal/Components/DisposalUnitComponent.cs

```diff
@@ -1,13 +1,45 @@
-﻿using Content.Shared.Disposal.Components;
-using Content.Shared.DragDrop;
-using Robust.Shared.GameObjects;
+using Content.Shared.Disposal.Components;
+using Robust.Client.Animations;
+using Robust.Shared.Audio;
 
 namespace Content.Client.Disposal.Components
 {
     [RegisterComponent]
     [ComponentReference(typeof(SharedDisposalUnitComponent))]
     public sealed class DisposalUnitComponent : SharedDisposalUnitComponent
     {
+        [DataField("state_anchored", required: true)]
+        public readonly string? StateAnchored;
+
+        [DataField("state_unanchored", required: true)]
+        public readonly string? StateUnAnchored;
+
+        [DataField("state_charging", required: true)]
+        public readonly string? StateCharging;
+
+        [DataField("overlay_charging", required: true)]
+        public readonly string? OverlayCharging;
+
+        [DataField("overlay_ready", required: true)]
+        public readonly string? OverlayReady;
+
+        [DataField("overlay_full", required: true)]
+        public readonly string? OverlayFull;
+
+        [DataField("overlay_engaged", required: true)]
+        public readonly string? OverlayEngaged;
+
+        [DataField("state_flush", required: true)]
+        public readonly string? StateFlush;
```

Comment:
These should be moved to layer maps in the prototype sprite component (you can put the enum at the end of this file)

Reply:
I _think_ I've done this the best way? This component was trying to map multiple strings (e.g., StateAnchored,StateUnAnchored,StageCharging) unto the same layer. Because there's no standard naming convention for the disposal unit RSI states (like, e.g. doors) The only nice way I could see to do that was to make the sprite have a layer for each thing the logic wanted to enable. Maybe there's a better way to do this?

Also, wasn't sure what to do with StateFlush - that string is used for initializing an animation, so maybe it can stay where it is?

When testing this, I noticed that the disposal units never got into the 'Charging' state (even on the `master` branch) so I've applied a bonus bugfix there.

Reply:
I dont see why StateFlush is special, `LayetGetState()` returns a state id that animations can use.
Also, you can rename the rsi states if you think its making things uglier, though making them different layers is fine imo